### PR TITLE
feat: enable the ability to sort the which key

### PR DESCRIPTION
**Pull request by abhaysp95**
Closes https://github.com/sxyazi/yazi/issues/652 - @taylo2allen

* sort "which" window in following way:
    - sort_by: "key|desc|none"
    - sort_sensitive: true|false
    - sort_reverse: true|false

### CODE REVIEW

**Comment by Rolv-Apneseth** [comment]
You should do a quick format on this - `cargo +nightly fmt`

**Comment by abhaysp95** [comment]
Done the formatting changes

**Comment by sxyazi** [comment]
Thank you for the PR! @abhaysp95 

I've made some little changes, would you like to give it a review?

File: yazi-core/src/which/sorter.rs [hunk 1]
```diff
@@ -9,30 +11,31 @@ pub struct WhichSorter {
...
-				let a = a.on.iter().map(|c| c.to_string()).collect::<String>();
-				let b = b.on.iter().map(|c| c.to_string()).collect::<String>();
-				by_alphabetical(&a, &b)
-			}),
-			SortBy::Desc => items.sort_unstable_by(|a, b| {
-				// what if description isn't present (need to check if it's mandatory or not)
-				// in case if it is not present, should I just panic ?
-				by_alphabetical(a.desc.as_ref().unwrap(), b.desc.as_ref().unwrap())
-			}),
+		let mut new = Vec::with_capacity(indices.len());
```
Comment (Rolv-Apneseth) [inline yazi-core/src/which/sorter.rs:35]: What do you think about this:

```rust
*items = indices.into_iter().map(|i| mem::take(&mut items[i])).collect();
```

**Comment by Rolv-Apneseth** [comment]
Does the sorting work for you? I think commit b44452d breaks it for me.

File: yazi-core/src/which/sorter.rs (hunk 1 above)
Comment (sxyazi) [reply yazi-core/src/which/sorter.rs:35]: good idea, will apply this suggestion

**Comment by sxyazi** [comment]
> Does the sorting work for you? I think commit [b44452d](https://github.com/sxyazi/yazi/commit/b44452dbd914142d1ef5b5edfd37716e23e2d2d2) breaks it for me.

Hi, I pushed a new commit should fix this.

**Comment by Rolv-Apneseth** [comment]
Nice, that fixed it

**Comment by abhaysp95** [comment]
> Thank you for the PR! @abhaysp95
> 
> I've made some little changes, would you like to give it a review?

It looks like I wrote the code in a verbose way. I tried to follow the example of how it was implemented for FileSorter. Maybe I thought that this will help in being more open to future changes. But I don't have any such problems with the change. All fine by me.

**Comment by sxyazi** [comment]
Thanks for the review, alright let me merge it!

**Pull request merged**